editoast/src/views/layers/geo_json.rs

```diff
@@ -0,0 +1,50 @@
+use mvt::GeomType;
+
+use rocket::serde::json::Value as JsonValue;
+use serde::{Deserialize, Deserializer};
+
+/// GeoJson used by the application
+pub enum GeoJsonType {
+    Point,
+    MultiPoint,
+    LineString,
+    MultiLineString,
+}
+
+impl GeoJsonType {
+    /// Gets MVT geom type corresponding to the GeoJson type
+    pub fn as_mvt_geom_type(&self) -> GeomType {
+        match self {
+            GeoJsonType::Point => GeomType::Point,
+            GeoJsonType::MultiPoint => GeomType::Point,
+            GeoJsonType::LineString => GeomType::Linestring,
+            GeoJsonType::MultiLineString => GeomType::Linestring,
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for GeoJsonType {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        Ok(match s.as_str() {
+            "Point" => GeoJsonType::Point,
+            "MultiPoint" => GeoJsonType::MultiPoint,
+            "LineString" => GeoJsonType::LineString,
+            "MultiLineString" => GeoJsonType::MultiLineString,
+            value => panic!("Unsupported MVT geom_type {value}"),
+        })
+    }
+}
+
+/// GeoJson representation
+#[derive(Deserialize)]
+pub struct GeoJson {
+    #[serde(rename = "type")]
+    pub geo_type: GeoJsonType,
+    #[serde(rename = "crs")]
+    _crs: JsonValue,
+    pub coordinates: JsonValue,
+}
```

Comment:
You should use polymorphism here.

```rust
#[derive(Clone, Serialize, Deserialize)]
#[serde(tag = "type", deny_unknown_fields)]
enum GeoJson {
    LineString {
       coordinates: Vec<(f64, f64)>
       crs: Option<...>
    },
    // ...
}
```

Reply:
this is so cool, like removing 50 lines and way clearer :open_mouth: :rocket: